### ISSUE TITLE
Filter steps give unexpected results with Java Boolean objects

### ISSUE THREAD

**Issue by rbeckman-nextgen** [nextgenhealthcare/connect#3166]
This is actually a bug in both 2.x and 3.x, but in different ways.

Say you have the two following rules joined by AND:

`return new java.lang.Boolean(false);`
`return true;`

In both 2.x and 3.x the message will not be filtered, when instead it should. However if you have only a single rule:

`return new java.lang.Boolean(false);`

In 2.x that will filter the message as expected, but in 3.x it will _not_ filter the message. That's because in 2.x we were just returning the object verbatim, but in 3.x we're always converting to a JavaScript primitive first. When you use a conditional expression and one of the operands is a Java Boolean, that is treated as a NativeJavaObject, and so its value is truthy regardless of whether the Java value is true or false. The exception is the == operator; Rhino explicitly checks for Java Boolean objects and unwraps them as necessary.

Basically around each filter rule we should be first testing them with " == true". And then, there would be no need for the "!!" toPrimitive operation.

Imported Issue. Original Details:
Jira Issue Key: MIRTH-3255
Reporter: narupley
Created: 2014-04-29T11:23:19.000-0700

**Comment by rbeckman-nextgen** [comment]
Fixed the bug by updating the javascript rule builder to check the truthiness returned by each user-defined javascript rule.

Imported Comment. Original Details:
Author: jaysenp
Created: 2014-05-28T18:38:47.000-0700

**Comment by rbeckman-nextgen** [comment]
Verified this no longer happens and filters act correctly whether a primitive JavaScript boolean or a Java Boolean is returned.

Imported Comment. Original Details:
Author: narupley
Created: 2014-08-27T09:10:28.000-0700
